### PR TITLE
Update EOL versions in `main`

### PR DESCRIPTION
**Pull request by richlander**
- Centos Stream 8 will be EOL soon.
- Debian 10 is long EOL.

Context: 
- https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/
- https://wiki.debian.org/DebianReleases

### CODE REVIEW

File: eng/common/templates/jobs/source-build.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ parameters:
   # This is the default platform provided by Arcade, intended for use by a managed-only repo.
   defaultManagedPlatform:
     name: 'Managed'
-    container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8'
+    container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0'
```
Comment (MichaelSimons) [inline eng/common/templates/jobs/source-build.yml:17]:
```suggestion
    container: 'mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream9'
```

**Review by MichaelSimons — DISMISSED** [review]
source build changes LGTM

File: eng/common/templates/job/source-index-stage1.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 parameters:
   runAsPublic: false
-  sourceIndexPackageVersion: 1.0.1-20240129.2
+  sourceIndexPackageVersion: 1.0.1-20240320.1
```
Comment (MichaelSimons) [inline eng/common/templates/job/source-index-stage1.yml:3]: I'm not familiar with this.  Why was this change made?

**Review by MichaelSimons — APPROVED** [review]

**Review by MichaelSimons — APPROVED** [review]

**Review by riarenas — APPROVED** [review]

**Pull request merged**